sched/task/exit.c

```diff
@@ -68,12 +67,6 @@ void _exit(int status)
   group_kill_children(tcb);
 #endif
 
-#if !defined(CONFIG_DISABLE_PTHREAD) && !defined(CONFIG_PTHREAD_MUTEX_UNSAFE)
```

Comment:
removing this introduce some flow change. I just want to understand if this is expected.
Before this change the `pthread_mutex_inconsistent(tcb);` was calling not depending on `TCB_FLAG_EXIT_PROCESSING`, but now after moving it to `nxtask_recover` the `TCB_FLAG_EXIT_PROCESSING` flag check start standing in the middle
```
void nxtask_exithook(FAR struct tcb_s *tcb, int status, bool nonblocking)
{
  /* Under certain conditions, nxtask_exithook() can be called multiple
   * times.  A bit in the TCB was set the first time this function was
   * called.  If that bit is set, then just exit doing nothing more..
   */

  if ((tcb->flags & TCB_FLAG_EXIT_PROCESSING) != 0)
    {
      return;
    }
...
  nxtask_recover(tcb);
}
```

Reply:
TCB_FLAG_EXIT_PROCESSING is only set at the end of this function:
```
void nxtask_exithook(FAR struct tcb_s *tcb, int status, bool nonblocking)
{
  /* Under certain conditions, nxtask_exithook() can be called multiple
   * times.  A bit in the TCB was set the first time this function was
   * called.  If that bit is set, then just exit doing nothing more..
   */

  if ((tcb->flags & TCB_FLAG_EXIT_PROCESSING) != 0)
    {
      return;
    }

  /* If the task was terminated by another task, it may be in an unknown
   * state.  Make some feeble effort to recover the state.
   */

  nxtask_recover(tcb);

  /* This function can be re-entered in certain cases.  Set a flag
   * bit in the TCB to not that we have already completed this exit
   * processing.
   */

  tcb->flags |= TCB_FLAG_EXIT_PROCESSING;
}
```
which ensure that nxtask_recover is executed only once.

Reply:
I mean that now the `pthread_mutex_inconsistent()` can be executed multiple times, but after this change it is executed only once. Just want to confirm that this verified.

Reply:
no, it just execute once since TCB_FLAG_EXIT_PROCESSING is checked.

Reply:
Yes, but `nxtask_exithook` is called from `nxtask_terminate` that is called from:
- `nxtask_exit`
- `nxtask_delete`
So now `pthread_mutex_inconsistent` will be called from that context as well.
Additionally `nxtask_recover` is called from `nxtask_restart`, so `pthread_mutex_inconsistent` will be called there as well.
Just want to make sure that all that cases are impacted intentionally and have been verified.

Reply:
> Yes, but `nxtask_exithook` is called from `nxtask_terminate` that is called from:
> 
> * `nxtask_exit`
> * `nxtask_delete`
>   So now `pthread_mutex_inconsistent` will be called from that context as well.
>   Additionally `nxtask_recover` is called from `nxtask_restart`, so `pthread_mutex_inconsistent` will be called there as well.

Yes, since all above functions should release the mutex which forget to release.

>   Just want to make sure that all that cases are impacted intentionally and have been verified.

Sure, it's always good to ensure all side effect get confirmed.

Reply:
Yes. I think that initial approach was covering `pthread` cases only, but this enables usage of `pthread_mutex` on task level.

Reply:
I mean that "Impact" is stated as "Code refactor", but in fact it is change of functionality

Reply:
> Yes. I think that initial approach was covering `pthread` cases only, but this enables usage of `pthread_mutex` on task level.

Yes, all pthread api can be used in main thread too.